airbyte-integrations/connectors/source-recharge/source_recharge/manifest.yaml

```diff
@@ -0,0 +1,269 @@
+    record_selector:
+      $ref: "#/definitions/selector"
+    requester:
+      $ref: "#/definitions/requester_deprecated_api"
+    paginator:
+      $ref: "#/definitions/paginator_deprecated_api"
+  # RETRIEVER FOR `MODERN API`
+  retriever_api_modern:
+    description: >-
+      Default Retriever for Modern API `2021-11` Full Refresh streams.
+    record_selector:
+      $ref: "#/definitions/selector"
+    requester:
+      $ref: "#/definitions/requester_modern_api"
+    paginator:
+      $ref: "#/definitions/paginator_modern_api"
+    # we should ignore all other req.params once we have the `next_page_token` in response
+    # for pagination in `2021-11` - modern api.
+    ignore_stream_slicer_parameters_on_paginated_requests: true
+  # RETRIEVER FOR `METAFIELDS` STREAM
+  retriever_metafields:
+    $ref: "#/definitions/retriever_api_modern"
+    partition_router:
+      type: ListPartitionRouter
+      cursor_field: owner_resource
+      values:
+        - address
+        - order
+        - charge
+        - customer
+        - store
+        - subscription
+      request_option:
+        inject_into: request_parameter
+        type: RequestOption
+        field_name: owner_resource
+
+  # BASE STREAMS
+  # FULL-REFRESH
+  base_stream:
+    primary_key: "id"
+    schema_loader:
+      $ref: "#/definitions/schema_loader"
+  base_deprecated_api_stream:
+    $ref: "#/definitions/base_stream"
+    retriever:
+      $ref: "#/definitions/retriever_api_deprecated"
+  base_modern_api_stream:
+    $ref: "#/definitions/base_stream"
+    retriever:
+      $ref: "#/definitions/retriever_api_modern"
+  # INCREMENTAL
+  base_incremental_stream:
+    $ref: "#/definitions/base_modern_api_stream"
+    incremental_sync:
+      type: CustomIncrementalSync
+      class_name: source_recharge.components.datetime_based_cursor.RechargeDateTimeBasedCursor
+      cursor_field: "updated_at"
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%dT%H:%M:%S%z"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+      start_time_option:
+        type: RequestOption
+        field_name: "updated_at_min"
+        inject_into: request_parameter
+
+  # FULL-REFRESH STREAMS
+  # COLLECTIONS
+  collections_stream:
+    description: >-
+      Collections Stream: https://developer.rechargepayments.com/2021-11/collections/collections_list
+    $ref: "#/definitions/base_modern_api_stream"
+    $parameters:
+      name: "collections"
+      data_path: "collections"
+  # METAFIELDS
+  metafields_stream:
+    description: >-
+      Metafields Stream: https://developer.rechargepayments.com/2021-11/metafields
+    $ref: "#/definitions/base_modern_api_stream"
+    retriever:
+      $ref: "#/definitions/retriever_metafields"
+    $parameters:
+      name: "metafields"
+      data_path: "metafields"
+  # PRODUCTS
+  products_stream:
```

Comment:
Differences observed:
* The current implementation slices the data using `updated_at`. The low-code does not. I'm very confused because [the documentation does not mention the query param `updated_at` for this API](https://developer.rechargepayments.com/2021-01/products/products_list) even though both the current version and the new one return the same number of records... In any case, I think I'm good with this change.

Reply:
Hmm, I've just checked the `products` stream `request` made to the API:

`'https://api.rechargeapps.com/products?limit=250'` and I don't see any `updated_at` passed or using.

Where did you see it?

Reply:
The low-code version indeed does `https://api.rechargeapps.com/products?limit=250` but the version `docker run -v $(pwd)/secrets:/data airbyte/source-{_SOURCE_NAME} read --config /data/config.json --catalog /data/configured_catalog.json --debug` provides the following outbound API calls:
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-01-01+00%3A00%3A01&updated_at_max=2023-01-31+00%3A00%3A01
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-01-31+00%3A00%3A02&updated_at_max=2023-03-02+00%3A00%3A02
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-03-02+00%3A00%3A03&updated_at_max=2023-04-01+00%3A00%3A03
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-04-01+00%3A00%3A04&updated_at_max=2023-05-01+00%3A00%3A04
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-05-01+00%3A00%3A05&updated_at_max=2023-05-31+00%3A00%3A05
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-05-31+00%3A00%3A06&updated_at_max=2023-06-30+00%3A00%3A06
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-06-30+00%3A00%3A07&updated_at_max=2023-07-30+00%3A00%3A07
* https://api.rechargeapps.com/products?limit=250&updated_at_min=2023-07-30+00%3A00%3A08&updated_at_max=2023-08-29+00%3A00%3A08

Reply:
Please make sure you don't have any leftovers from the previous builds for `source-recharge`, at least it's weird why it formatted like this, because the products or similar streams would return the error for `invalid_query_param` if any of `updated_at_min` is applied to the real request.

Reply:
So, it turns out that this comes from the `1.1.6`, and I suspect even earlier. 

Because of:
-  https://github.com/airbytehq/airbyte/blob/58eddccace538bf5248d4d1d035132ca72a62351/airbyte-integrations/connectors/source-recharge/source_recharge/api.py#L114-L120
and
- https://github.com/airbytehq/airbyte/blob/58eddccace538bf5248d4d1d035132ca72a62351/airbyte-integrations/connectors/source-recharge/source_recharge/api.py#L137-L141

I didn't realize it makes the requests with more params than needed, but I suspect it is OK to have, because it doesn't cause the issues on the API side. 

So, we have this fixed for the migrated version!

Reply:
The fact that both version works flabbergast me! But yeah, I'm more comfortable with the low-code solution we have right now. Thanks for making sure this change was intended and wanted